corehq/apps/users/models.py

```diff
@@ -1169,16 +1170,16 @@ def user_session_data(self):
         })
         return session_data
 
-    def delete(self, deleted_by, deleted_via=None):
+    def delete(self, deleted_by_domain, deleted_by, deleted_via=None):
         self.clear_quickcache_for_user()
         try:
             user = self.get_django_user()
             user.delete()
-            if deleted_by:
-                log_model_change(deleted_by, user, message=f"deleted_via: {deleted_via}",
-                                 action=ModelAction.DELETE)
         except User.DoesNotExist:
             pass
+        if deleted_by:
```

Comment:
similar comment to on the other PR, is this check just for tests? Should it validate that `deleted_by` is None only in tests.

Another thought, if this is being skipped in tests what's the point of passing in the domain in all the test usages? Why not just pass in `None`

Reply:
> similar comment to on the other PR

adding for reference, https://github.com/dimagi/commcare-hq/pull/29902#discussion_r650870658

> is this check just for tests? Should it validate that deleted_by is None only in tests.

yes. adding a check to ensure deleted_by is only for tests.


> Another thought, if this is being skipped in tests what's the point of passing in the domain in all the test usages? Why not just pass in None

You could pass None yes.
I think I went with domain because the original intention was to "avoid logging in tests to avoid DB overhead" and not to "allow tests to not pass domain".

Open to suggestions on this. 

Honestly, 
When i was fixing tests and passing domain, it seemed okay to do so but when it was too many files and i was regretting it, it was too late, so I just continued on it. Was thinking that I could have just updated the method signature to have `domain=None` and not worry about it in tests.

Once done, was thinking about positives,
1. that there was not one more `None` next to `deleted_by=None` in tests, which was nicer to read.
2. this built a sense that domain is to be passed on user actions now and get that as a habit, which is going to be new for everyone.

Reply:
I think it's fine as is and no need to go back and change all places to pass None

Reply:
What would you suggest the ideal approach would have been here?

Reply:
probably just passing None in tests